### PR TITLE
Added TypeScript type definitions

### PR DESCRIPTION
**Pull request by WhySoBad**
I added an index.d.ts file.
This allows to get intellisense support for regular usage with JavaScript and to use this library with TypeScript using the type definitions.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+export interface CorsairDevice {
+  type: number
+  model: string
+  physicalLayout: number
+  logicalLayout: number
+  capsMask: number
+  ledsCount: number
+  channels: {
+    channelsCount: number
```
Comment (intrueder) [inline index.d.ts:145]: This does not look complete

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+export interface CorsairProtocolHandshake {
+  sdkVersion: string
+  serverVersion: string
+  sdkProtocolVersion: number
+  serverProtocolVersion: number
+  breakingChanges: boolean
+}
+
+export interface CorsairDevice {
```
Comment (intrueder) [inline index.d.ts:137]: Should be named `CorsairDeviceInfo`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+export function CorsairGetLedIdForKeyName(key: string): number
+
+/**
+ * Function to request explusive control
+ *
+ * @returns void
+ */
+
+export function CorsairRequestControl(): void
```
Comment (intrueder) [inline index.d.ts:343]: Returns `boolean`, see https://corsairofficial.github.io/cue-sdk/#corsairrequestcontrol

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+/**
+ * Function to release exclusive requested control
+ *
+ * @returns void
+ */
+
+export function CorsairReleaseControl(): void
```
Comment (intrueder) [inline index.d.ts:351]: Returns `boolean`

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+export function CorsairSubscribeForEvents(
+  callback: (event: {
+    id: 'macrokeydown' | 'macrokeyup' | 'deviceconnect' | 'devicedisconnect'
+    deviceId: string
+  }) => void
+): boolean
+
+/**
+ * Function to unsubscribe to events
```
Comment (intrueder) [inline index.d.ts:398]: to events -> from events

**Review by intrueder — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (WhySoBad) [reply index.d.ts]: In the [CUE SDK documentation](https://corsairofficial.github.io/cue-sdk/#corsairdeviceinfo) is a property deviceId for the CorsairDeviceInfo but I didn't manage to get this with any of my devices. Does this property really exist?

Comment (intrueder) [reply index.d.ts]: I'll create an issue for this, js wrapper does not have CorsairDeviceInfo.deviceId property but it does exist in the CUESDK.

Comment (intrueder) [reply index.d.ts]: #15 will fix it once this PR is merged

**Review by intrueder — APPROVED** [review]
LGTM

**Pull request merged**